jib-core/src/main/java/com/google/cloud/tools/jib/configuration/DockerHealthCheck.java

```diff
@@ -29,13 +30,16 @@
   /** Builds the immutable {@link DockerHealthCheck}. */
   public static class Builder {
 
-    @Nullable private final ImmutableList<String> command;
+    private final ImmutableList<String> command;
     @Nullable private Duration interval;
     @Nullable private Duration timeout;
     @Nullable private Duration startPeriod;
     @Nullable private Integer retries;
 
-    private Builder(@Nullable ImmutableList<String> command) {
+    private Builder(ImmutableList<String> command) {
+      Preconditions.checkArgument(
+          command.size() > 1 && !command.contains(""),
+          "command must not be empty and must not contain empty tokens");
```

Comment:
For conditions joined with `&&`, prefer separating them. It helps identify which condition has been violated.

And maybe this should be `!command.isEmpty()` or the error message needs to change?

Reply:
It needs to check if size is >1, not isEmpty, because "CMD" or "CMD-SHELL" is added before it, and a command containing "CMD"/"CMD-SHELL" with no other arguments isn't valid.

Reply:
Should we be checking if `command.stream().anyMatch(str -> str.trim().isEmpty)` or something? Not just against the empty string but the whitespace string?

Reply:
Docker allows it.

Dockerfile:
```Dockerfile
HEALTHCHECK CMD [" "]
```

`docker inspect`:

```
"Healthcheck": {
  "Test": [
    "CMD",
    " "
  ]
},
```

Reply:
Ah wait, it takes empty strings too for CMD, just not CMD-SHELL. I think I have to move the precondition checks.